dex/src/main/scala/com/wavesplatform/dex/api/http/routes/MatcherApiRoute.scala

```diff
@@ -1033,6 +1037,55 @@ class MatcherApiRoute(
     complete { Json.toJson(orderDb.transactionsByOrder(orderId)) }
   }
 
+  @Path("/debug/config")
+  @ApiOperation(
+    value = "Returns current matcher's configuration",
+    httpMethod = "GET",
+    authorizations = Array(new Authorization(SwaggerDocService.apiKeyDefinitionName)),
+    tags = Array("debug"),
+    response = classOf[HttpResponse]
+  )
+  def getConfig: Route = (path("config") & get & withAuth)  {
+    complete {
+
+      var outputConfig = config.getObject("waves").toConfig
+
+      def pathShouldNotBeDeleted(path: String): Boolean = ("(user|pass|seed|private)".r findFirstIn path).isEmpty
+
+      def modifyConfig(path: String, config: Config): Unit = {
+        val itr = config.root().entrySet().iterator()
+
+        while(itr.hasNext()){
+
+          val key = itr.next().getKey
+          val absolutePath = if(path.length > 0) s"$path.$key" else key
+
+          config.root().get(key).valueType() match {
+            case ConfigValueType.OBJECT => {
+              if(pathShouldNotBeDeleted(absolutePath)) {
+                if(config.getObject(key).size() == 0) {
+                  outputConfig = outputConfig.withoutPath(absolutePath)
+                } else {
+                  modifyConfig(absolutePath, config.getObject(key).toConfig)
+                }
+              }
+            }
+            case _ => if(!pathShouldNotBeDeleted(absolutePath)) outputConfig = outputConfig.withoutPath(absolutePath)
+          }
+        }
+      }
+
+      modifyConfig("", outputConfig)
```

Comment:
I think this won't modify `outputConfig`, from the documentation:
```
    /**
     * Clone the object with the given key removed.
     *
     * @param key
     *            key to remove
     * @return a copy of the object minus the specified key
     */
    ConfigObject withoutKey(String key);
```

So this function returns a copy of object. And we don't use this copy